pandas/io/formats/csvs.py

```diff
@@ -120,58 +163,88 @@ def __init__(
         # and make sure sure cols is just a list of labels
         cols = self.obj.columns
         if isinstance(cols, ABCIndexClass):
-            cols = cols.to_native_types(
-                na_rep=na_rep,
-                float_format=float_format,
-                date_format=date_format,
-                quoting=self.quoting,
-            )
+            return cols.to_native_types(**self._number_format)
         else:
-            cols = list(cols)
+            assert isinstance(cols, Sequence)
+            return list(cols)
 
-        # save it
-        self.cols = cols
+    @property
+    def _number_format(self) -> dict:
```

Comment:
technically this should be Dict[str, Any]

Reply:
cc @simonjayhawkins don't we have a mypy setting that fails on this? (or should be have a code-check)?

Reply:
Thank you! Corrected typing.

Reply:
> cc @simonjayhawkins don't we have a mypy setting that fails on this? (or should be have a code-check)?

see #30539